src/vg/civcraft/mc/namelayer/listeners/MercuryMessageListener.java

```diff
@@ -49,5 +88,29 @@ else if (reason.equals("transfer")){
 			gm.invalidateCache(g.getName());
 		}
 	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogin(PlayerJoinEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "login "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGIN");
+	}
+	
+	@EventHandler(priority = EventPriority.HIGHEST)
+	public void onLogoff(PlayerQuitEvent event){
+		MercuryPlugin.handler.sendMessage("all", "name_layer", "logoff "+event.getPlayer().getDisplayName());
+		Bukkit.getLogger().info("LOGOFF");
+	}
 
+	@Override
+	public void run() {
+		NameLayerPlugin nl = NameLayerPlugin.getInstance();
+		nl.isMercuryEnabled = Bukkit.getPluginManager().isPluginEnabled("Mercury");
+		if (nl.isMercuryEnabled){
+			nl.getServer().getPluginManager().registerEvents(this, nl);
```

Comment:
I copied and pasted my bad. Just register the class.
On Jun 5, 2015 12:01 PM, "Darius Scott" <notifications@github.com> wrote:

> In src/vg/civcraft/mc/namelayer/listeners/MercuryMessageListener.java
> <https://github.com/Civcraft/NameLayer/pull/112#discussion_r31825624>:
>
> >
> > +	@Override
> > +	public void run() {
> > +		NameLayerPlugin nl = NameLayerPlugin.getInstance();
> > +		nl.isMercuryEnabled = Bukkit.getPluginManager().isPluginEnabled("Mercury");
> > +		if (nl.isMercuryEnabled){
> > +			nl.getServer().getPluginManager().registerEvents(this, nl);
>
> You want it to make another MercuryMessageListener? How come? Wouldn't
> nl.getServer().getPluginManager().registerEvents(this, nl); increment the
> reference so it wouldnt get gc'd when the runnable is over?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Civcraft/NameLayer/pull/112/files#r31825624>.
>